tests/ui/parser/missing-fat-arrow.stderr

```diff
@@ -0,0 +1,78 @@
+error: expected identifier, found keyword `self`
+  --> $DIR/missing-fat-arrow.rs:8:13
+   |
+LL |         Some(x) if x == y {
+   |                         - while parsing this struct
+LL |             self.next_token()?;
+   |             ^^^^ expected identifier, found keyword
+   |
+help: you might have meant to start a match arm after the match guard
+   |
+LL |         Some(x) if x == y => {
+   |                           ++
+
+error: expected one of `,`, `:`, or `}`, found `.`
+  --> $DIR/missing-fat-arrow.rs:25:14
+   |
+LL |         Some(a) if a.value == b {
+   |                               - while parsing this struct
+LL |             a.value = 1;
+   |             -^ expected one of `,`, `:`, or `}`
+   |             |
+   |             while parsing this struct field
+   |
+help: try naming a field
+   |
+LL |             a: a.value = 1;
+   |             ++
```

Comment:
I don't think that's helpful, given that `=` has type `()`

Reply:
The parser isn't doing arbitrary lookup to figure out what the rest of the stream is, at this point all the parser knows is that we had an ident followed by a dot.

Reply:
Right, makes sense.